### PR TITLE
Add SECURITY.md for vulnerability reporting

### LINKED ISSUE #4191: Create SECURITY.md
Hello 👋

I run a security community that finds and fixes vulnerabilities in OSS. A researcher (@yuligesec) has found a potential issue, which I would be eager to share with you.

Could you add a `SECURITY.md` file with an e-mail address for me to send further details to? GitHub [recommends](https://docs.github.com/en/code-security/getting-started/adding-a-security-policy-to-your-repository) a security policy to ensure issues are responsibly disclosed, and it would help direct researchers in the future.

Looking forward to hearing from you 👍

(cc @huntr-helper)

### PR DESCRIPTION
**Pull request by tonyhoo**
*Issue #, if available:*
Resolves #4191
*Description of changes:*
Add SECURITY.md for vulnerability reporting

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Security Policy
+
+## Supported Versions
+
+| Version       | Supported          |
+| ------------- | ------------------ |
+| 1.1.1         | :white_check_mark: |
+| < 1.1.1       | :x:                |
```
Comment (suzhoum) [inline .github/SECURITY.md:8]: Does this mean that #1-3 below are only supported in 1.1.1? Should we move this table following the next session `How we do security`?

File: .github/SECURITY.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Security Policy
```
Comment (Innixma) [inline .github/SECURITY.md:1]: Why are we putting SECURITY.md into `.github/`? Scikit-learn has it in their root directory.

File: .github/SECURITY.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+
+As much as possible, AutoGluon relies on automated tools to do security scanning. In particular, we support:
+
+1. Dependency Analysis: Using Dependabot
+2. Docker Scanning: Using Snyk
+3. Code Analysis: Using CodeGuru
+
+
+## Vulnerability Reporting
```
Comment (Innixma) [inline .github/SECURITY.md:20]: nit: "Reporting a Vulnerability" reads slightly nicer imo. (that is the wording used by scikit-learn)

File: .github/SECURITY.md (hunk 1 above)
Comment (tonyhoo) [reply .github/SECURITY.md:8]: This means only 1.1.1 will be applicable for security patches

File: .github/SECURITY.md (hunk 2 above)
Comment (tonyhoo) [reply .github/SECURITY.md:1]: Will move it to root

File: .github/SECURITY.md (hunk 3 above)
Comment (tonyhoo) [reply .github/SECURITY.md]: updated

**Review by Innixma — APPROVED** [review]
LGTM!

**Pull request merged**